carts/views.py

```diff
@@ -0,0 +1,62 @@
+import json
+
+from django.http import JsonResponse
+from django.views import View
+from django.db.models import Q
+
+from carts.models import Cart
+from users.models import User
+from products.models import Product
+from users.decorator import login_decorator
+
+class CartView(View):
+    @login_decorator
+    def post(self, request):
+        try:
+            data       = json.loads(request.body)
+            product_id = data['product_id']
+            quantity   = data['quantity']
+
+            if not Product.objects.filter(id=product_id).exists():
+                return JsonResponse({"MESSAGE":"DOES_NOT_EXIST_ERROR"}, status=400)
+
+            if Cart.objects.filter(product_id=product_id).exists():
+                return JsonResponse({"MESSAGE":"ALREADY_EXIST"}, status=400)
+
+            Cart.objects.create(
+                user       = request.user,
+                product_id = product_id,
+                quantity   = quantity,
+                )
+
+            return JsonResponse({'MESSAGE':'CREATE'}, status=201)
+
+        except KeyError:
+            return JsonResponse({'MESSAGE':'KEY_ERROR'}, status=400)
+
+    @login_decorator
+    def patch(self, request):
+        try:
+            data       = json.loads(request.body)
+            product_id = data['product_id']
+            quantity   = data['product_quantity']
+            cart       = Cart.objects.filter(product_id=product_id)
+            cart.update(quantity=quantity)
+
+            return JsonResponse({'MESSAGE':'SUCCESS'}, status=201)
+
+        except KeyError:
+            return JsonResponse({'MESSAGE':'KEY_ERROR'}, status=400)
+
+    @login_decorator
+    def delete(self, request):
+        cart = Cart.objects.filter(Q(user=request.user) & Q(product_id=request.GET.get('product_id', None)))
+        cart.delete()
+        return JsonResponse({'MESSAGE':'SUCCESS'}, status=201)
```

Comment:
1. RestAPI에서는 DELETE 구현시 `path parameter`를 사용해서 구현해주셔야 합니다!
`DELETE cart/1`
2. delete 성공시 status code는 204입니다~

```suggestion
    @login_decorator
    def delete(self, request, cart_id):
        Cart.objects.filter(user=request.user, id=cart_id).delete()
        
        return JsonResponse({'MESSAGE':'NO_CONTENT'}, status=204)
```

Reply:
path 파라미터로 수정하였습니다.